lib/modules/dosomething/dosomething_signup/includes/SignupTransformer.php

```diff
@@ -66,6 +66,46 @@ public function show($id) {
   }
 
   /**
+   * Create the specified resource.
+   *
+   * @param  array $parameters
+   * @return array
+   */
+  public function create($parameters) {
+    // @TODO: if "key" present, do stuff to check against the key and see if valid
+    $values = [
+      'uid' => $parameters['user'],
+      'nid' => $parameters['campaign'],
+      'run_nid' => $parameters['campaign_run'],
+      'source' => $parameters['source'],
+      'timestamp' => time(),
+      'transactionals' => NULL,
+    ];
+
+    try {
+      $record = (new SignupsController)->magick_create($values);
+
+      http_response_code('201');
+
+      $response = [
+        'success' => [
+          'message' => 'Signup successfully created! (But actually return the record created)',
+        ],
+      ];
+    } catch (Exception $error) {
+      http_response_code('409');
```

Comment:
I wonder if it'd make sense to use a 422 Unprocessable Entity validation error here for consistency with how we enforce uniqueness elsewhere? Here's the [status codes that we use on Northstar](https://github.com/DoSomething/northstar/tree/dev/documentation#errors--status-codes).